ibis/expr/types/arrays.py

```diff
@@ -725,7 +725,48 @@ def union(self, other: ir.ArrayValue) -> ir.ArrayValue:
         """
         return ops.ArrayUnion(self, other).to_expr()
 
-    def zip(self, other: ir.Array, *others: ir.Array) -> ir.Array:
+    def intersection(self, other: ArrayValue) -> ArrayValue:
```

Comment:
I have a slight preference to call both the API method and the operation it intersect instead of intersection.

Reply:
Ah, definitely! We should be consistent with `Table`, which def `intersect`